coral-spark/src/test/java/com/linkedin/coral/spark/CoralSparkTest.java

```diff
@@ -829,16 +829,6 @@ public void testSelectStar() {
     assertEquals(expandedSql, targetSql);
   }
 
-  @Test
-  public void testRedundantCastRemovedFromCaseCall() {
-    final String sourceSql = "SELECT CASE WHEN TRUE THEN NULL ELSE split(b, ' ') END AS col1 FROM complex";
-    String expandedSql = getCoralSparkTranslatedSqlWithAliasFromCoralSchema(sourceSql);
-
-    String targetSql =
-        "SELECT CASE WHEN TRUE THEN NULL ELSE split(complex.b, ' ') END col1\n" + "FROM default.complex complex";
```

Comment:
why was this UT deleted?

Reply:
This UT was added together with the transformation `removeCastToEnsureCorrectNullability`. After we remove it, the target SQL will be:
```
SELECT CASE WHEN TRUE THEN CAST(NULL AS ARRAY<STRING>) ELSE CAST(split(complex.b, ' ') AS ARRAY<STRING>) END col1
FROM default.complex complex
```
Because Calcite adds the explicit `CAST` to ensure the types match for the field `col1` in `CoralSqlNode1 -> Coral IR` conversion. The newly added `CAST` will not cause regression after we fix the Coral-Schema nullability issue in #369.

Reply:
not sure im following the description. 
For input SQL - "SELECT CASE WHEN TRUE THEN NULL ELSE split(b, ' ') END AS col1 FROM complex"
could you share - 
hiveSqlNode, coralSqlNode1, coralSqlNode2 representation?

When you say `The newly added CAST`, is it the same CAST(s) added on LHS or a different one?
Also, seems like we can retain this unit test

Reply:
HiveSqlNode & CoralSqlNode1:
```
SELECT CASE WHEN TRUE THEN NULL ELSE `split`(`b`, ' ') END AS `col1`
FROM `complex`
```
CoralSqlNode2:
```
SELECT CASE WHEN TRUE THEN CAST(NULL AS ARRAY<VARCHAR(65535) CHARACTER SET `ISO-8859-1`>) ELSE CAST(`split`(`complex`.`b`, ' ') AS ARRAY<VARCHAR(65535) CHARACTER SET `ISO-8859-1`>) END AS `col1`
FROM `default`.`complex` AS `complex`
```
`The newly added CAST` is the one added in `CoralSqlNode2` above by Calcite while converting `CoralSqlNode1` to `Coral IR`, so it exists in `CoralSqlNode2`.
I don't think we need to retain this unit test because there is no corresponding implementation in Coral after this PR.

Reply:
there are two CAST operators in the CoralSqlNode2 representation. One around `NULL` and the other around `split`(`b`, ' ')`.
Are both added in CoralRelNode -> CoralSqlNode2 layer due to the override `requireCastOnString()` in https://github.com/linkedin/coral/blob/master/coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java#L81?

Reply:
No, both are added while converting `CoralSqlNode1` to `Coral IR` in `HiveSqlToRelConverter`. `requireCastOnString` only ensures the `CAST` in `CAST(string AS type)` won't be dropped incorrectly.

Reply:
okay, so it exists in the RelNode representation? (1) How is this information captured in RelNode?
(2) Also, to avoid introducing disparity between CoralSqlNode1, 2, is it possible to add this change in CoralSqlNode1 as well?

Reply:
Thinking a bit, we might have a deeper issue here. What is the answer to this question: Does Coral SQL Node spec expect operand types to match or is it optional? If it expects them to match, then we need CoralSqlNode1 to have the `CAST`. If it is optional, then Trino still needs to transform to ensure they match. I think we have discussed this in the past.

Reply:
I think it's optional as Calcite is able to add explicit `CAST` if there is a type mismatch.
As for Trino case, it's because Trino SQL dialect requires it, it doesn't imply that CoralSqlNode must mandate it.

Reply:
Not clear if this answers my question. Are you saying that it is optional in Coral IR spec?

Reply:
I would advocate for CoralSqlNode to be as detailed as possible. This will enrich the SqlNode representation and possibly help with sqlNode transformations across different dialects.

Reply:
Yes, it's optional in coral IR spec because we can handle both during the conversion from SqlNode to RelNode.

Reply:
If it is optional in Coral SqlNode IR, then Trino must have a defense mechanism to re-instate it, right? Doing it in Sql-to-Rel is not an answer because that does not speak to the "Coral IR SqlNode spec".

Reply:
I think Trino doesn't need a defense mechanism, because explicit `CAST` is mandatary in Coral IR RelNode, it will definitely appear in the CoralSqlNode2 and the result Trino SQL.

Reply:
As mentioned, handing it in Coral RelNode IR is not the answer, because this compromises the SqlNode spec. In fact, with Coral SqlNode IR, going through the RelNode is optional.

Reply:
I see, thanks for explaining. Then I think we need to make types match mandatory in `Coral IR SqlNode spec`. I think we need to add this case to #357 as a follow-up work to make types match in CoralSqlNode1.
Besides, if we want to make `100% CoralSqlNode1 == CoralSqlNode2` for Coral IR SqlNode spec, we may also need to run a comparison test for all the production views to find all the cases where `CoralSqlNode1 != CoralSqlNode2` and fix them.

Reply:
After offline discussion, there might be a miscommunication in this thread. My comments where mostly pertaining to changes like #373. That said, I think changes in this PR (#377) are fine, and will follow up on this thread in #373.